LMR-74

Daria-pugacheva opened the pull request:
Сделала функционал отправки с фронта изменений данных юзера + добавила ФИО для нашего тестового юзера начального.

Написала в UserController TODO-шку: Надо откорректировать метод модификации пользователя. Там сейчас идет проверка, совпадают ли пароль и подтверждение пароля. А при модификации с фронта пароли не летят в составе ДТО и при этом хорошо бы проверять, не совпадает ли новый еmail с каким-то в базе. 
Т.о. в итоге для модификации пользователя все-таки надо доделать бэк (имеющийся метод недостаточно заточен под это)
Для ревью позвала Артема, т.к. , возможно, ему сразу удобно будет поправить метод, т.к. все равно будет смотреть, что сломалось в регистрации.

Daria-pugacheva committed:
Сделала функционал отправки с фронта изменений данных юзера + добавила ФИО для нашего тестового юзера начального.
Написала в UserController TODO-шку: Надо откорректировать метод модификации пользователя. Там сейчас идет проверка, совпадают ли пароль и подтверждение пароля. А при модификации с фронта пароли не летят в составе ДТО и при этом хорошо бы проверять, не совпадает ли новый еmail с каким-то в базе.

Daria-pugacheva committed:
Merge remote-tracking branch 'origin/develop' into LMR-74

Haazaad commented on source/core/src/main/java/ru/letmerent/core/controllers/UserController.java:
```diff
@@ -106,7 +106,13 @@ public ResponseEntity<?> modifyUser(@RequestBody UserDto userDto) {
         userService.saveUser(userDto);
         return new ResponseEntity<>(HttpStatus.OK);
-    }
+    } //TODO: Вопрос: этот метод, видимо надо модифицирвоать?:
```
Так мы сохраняем уже существующего пользователя, зачем такие сложности?

Daria-pugacheva replied:
Да, я уже сделала отдельный метод, т.к. этот заморочный. Я его закомментировала на случай, если кто-то зачем-то создавал

Haazaad commented on source/core/src/main/resources/db/changelog/changes/sql/v2_insert_initial_data.sql:
```diff
@@ -8,8 +8,8 @@ values ('Ручной', 'Ручной инструмент'),
 
-insert into users (first_name, email, username, password)
-values ('Test', 'test@test.com', 'test', '1111');
+insert into users (first_name, second_name, last_name, email, username, password)
```
Не заведется такое изменение без чистки коммитов liquibase в самой базе.
Лучше сделать необходимые изменения локально и не тянуть их в общий репозиторий

Daria-pugacheva replied:
хм... а есть ощущение, что у меня все получилось... :/ Интересно, почему... Так а если эти изменения нужны в общем репозитории, почему не сделать их в общем? Вот смотри, задача такая, чтобы я запустила проект на develop, запустила докер  и у меня уже в базе лежал этот пользователь со всеми заполненными полями.  Что я сделала не так? У нас ведь, по идее, когда я запускаю Докер и формирую свежий контейнер как бы, создается же из этих вот скриптов наполнение БД. Разве не так?

Daria-pugacheva committed:
На бэке в UserController и UserService сделала методы для обновления данных пользователя с проверкой, чтобы адрес эл.почты не совпадал ни с каким другим в базе.
Обнолвение данных пользователя работает полностью.

И важную вещь поправила в profile.js : сделала так, чтобы, если пользователь не авторизован, никакие запросы на бэк не летели из profile.js (ранее упустила из виду - мой косяк).

Daria-pugacheva committed:
Добавила проверку, что при совпадении email адрес эл.почты не принадлежит обновляемому юзеру (иначе при обновлении не получается оставить неизменным свой email)

Daria-pugacheva committed:
Поменяла формулировку ошибки при использовании чужого адреса эл.почты

Daria-pugacheva committed:
Поменяла формулировку ошибки при использовании чужого адреса эл.почты
